Assets/_Standard Assets/Characters/Scripts/Physics/OpenCharacterController.cs

```diff
@@ -1836,7 +1842,22 @@ bool CanStickToGround(Vector3 moveVector, out MoveVector getDownVector)
 				project.Normalize();
 
 				// Slide along the obstacle
-				moveVector = project * remainingDistance;
+				// If m_OnWallsKeepVelocity is enabled or the angle is under m_OnWallsMaxAngle, there will be no slowdown
+				var slowDownAngle = Vector3.Angle(project, moveVector);
```

Comment:
As mentioned in the main comment, a suggested revision:

Reply:
```
float slowDownFactor = 1.0f;

if (!m_OnWallsKeepVelocity)
{
	// Cosine of angle between the movement direction and the tangent is equivalent to the sin of
	// the angle between the movement direction and the normal, which is the sliding component of
	// our movement. 
	var cosine = Vector3.Dot(project, direction);
	var invRescaleFactor = 1 / Mathf.Cos(m_OnWallsMaxAngle * Mathf.Deg2Rad);
	slowDownFactor = Mathf.Clamp01(cosine * invRescaleFactor);
}

moveVector = project * (remainingDistance * slowDownFactor);
```

Reply:
This can be further optimised by precalculating `invRescaleFactor` in Awake() to take out the most expensive part of this calculation. (Probably also want to, in the editor only, recalculate it here so that a user can update the values live in the inspector, but we still get the performance benefit in builds).